### PR TITLE
Fix broken links and remove obselete ones

### PR DESCRIPTION
**Pull request by richardgreg**
**Description**

Fixed some broken links in the docs and removed obsolete ones. Also added helpful resources where applicable.

### CODE REVIEW

**Comment by richardgreg** [comment]
Hi @Nicca42, @smartcontracts and @OPMattie, I made substantial changes that I think would improve the Optimism docs. I'd appreciate it if someone looked at my PR and gave me some feedback. Thanks. 😊

File: src/docs/developers/bedrock/differences.md [hunk 1]
```diff
@@ -286,25 +286,25 @@ Note that in Bedrock there are two contracts:
...
   The `L1BlockNumber` contract provides the number of the latest L1 block. 
   In Bedrock it is simply a proxy to [`L1Block`](#l1block). 
 - [L2CrossDomainMessenger](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/L2/L2CrossDomainMessenger.sol):
   The `L2CrossDomainMessenger` contract is used to send messages from Optimism to Ethereum.
 - [L2StandardBridge](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/L2/L2StandardBridge.sol):
   The `L2StandardBridge` contract is used to "attach" assets (ETH and ERC-20 tokens) to messages that are then sent by `L2CrossDomainMessenger`.
-- [WETH9](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/L2/WETH9.sol): 
-  [The WETH9 contract](https://weth.io/) is an ERC-20 token that wraps around ETH to provide extra functionality, such as approvals.
+- [WETH9](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts/contracts/L2/predeploys/WETH9.sol):
```
Comment (smartcontracts) [inline src/docs/developers/bedrock/differences.md:296]:
```suggestion
- [WETH9](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/vendor/WETH9.sol): 
```

File: src/docs/developers/bedrock/differences.md [hunk 2]
```diff
@@ -286,25 +286,25 @@ Note that in Bedrock there are two contracts:
...
 #### Historical contracts
 
 These are contracts that are no longer relevant, but are kept as part of the state in case there is a call in any dapp that uses them.
 
-- [DeployerWhitelist](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/L2/DeployerWhitelist.sol):
+- [DeployerWhitelist](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/legacy/DeployerWhitelist.sol):
   The `DeployerWhitelist` contract used to manage the whitelist before [Optimism moved out of beta](https://twitter.com/optimismFND/status/1471571415774023682).
 
-- [OVM_ETH](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/L2/OVM_ETH.sol):
+- [OVM_ETH](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts/contracts/L2/predeploys/OVM_ETH.sol):
```
Comment (smartcontracts) [inline src/docs/developers/bedrock/differences.md:307]:
```suggestion
- [OVM_ETH](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/legacy/LegacyERC20ETH.sol):
```

**Review by smartcontracts — COMMENTED** [review]
Left some small comments, otherwise LGTM once fixed!

**Comment by richardgreg** [comment]
Changes applied

**Review by smartcontracts — APPROVED** [review]
LGTM!

**Comment by smartcontracts** [comment]
Great work, thank you!

**Comment by richardgreg** [comment]
> Great work, thank you!

Thanks 😊

**Pull request merged**